From dde7f3e67bc19d695cc2cb1726580784673ed228 Mon Sep 17 00:00:00 2001
From: Andre Przywara <andre.przywara@arm.com>
Date: Sat, 10 Feb 2024 23:19:32 -0500
Subject: [PATCH] arch: arm64: dts: allwinner: sun50i-h616: improve thermals

[    3.134284] OF: /thermal-zones/gpu-thermal/cooling-maps/map0: could not get #cooling-cells for /soc/gpu@1800000
[    3.134299] thermal_sys: Add a cooling_device property with at least one device
[    3.134304] thermal thermal_zone0: binding zone gpu-thermal with cdev cpufreq-cpu0 failed:-2

Signed-off-by: Andre Przywara <andre.przywara@arm.com>
Signed-off-by: Patrick Yavitz <pyavitz@xxxxx.com>
---
 .../arm64/boot/dts/allwinner/sun50i-h616.dtsi | 150 ++++++++----------
 1 file changed, 64 insertions(+), 86 deletions(-)

diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
index 18ab5585af..9696f8e228 100755
--- a/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
+++ b/arch/arm64/boot/dts/allwinner/sun50i-h616.dtsi
@@ -1069,6 +1069,19 @@ mdio1: mdio {
 			};
 		};
 
+		ths: thermal-sensor@5070400 {
+			compatible = "allwinner,sun50i-h616-ths";
+			reg = <0x05070400 0x400>;
+			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&ccu CLK_BUS_THS>;
+			clock-names = "bus";
+			resets = <&ccu RST_BUS_THS>;
+			nvmem-cells = <&ths_calibration>;
+			nvmem-cell-names = "calibration";
+			allwinner,sram = <&syscon>;
+			#thermal-sensor-cells = <1>;
+		};
+
 		usbotg: usb@5100000 {
 			compatible = "allwinner,sun50i-h616-musb",
 				     "allwinner,sun8i-h3-musb";
@@ -1469,24 +1482,6 @@ r_rsb: rsb@7083000 {
 			#size-cells = <0>;
 		};
 
-		ths: thermal-sensor@5070400 {
-			/* The Thermal Sensor Controller(THS) embeds four thermal sensors,
-			sensor0 is located in GPU
-			sensor1 is located in VE
-			sensor2 is located in CPU
-			sensor3 is located in DDR
-			*/
-			compatible = "allwinner,sun50i-h616-ths";
-			reg = <0x05070400 0x400>;
-			interrupts = <GIC_SPI 19 IRQ_TYPE_LEVEL_HIGH>;
-			clocks = <&ccu CLK_BUS_THS>;
-			clock-names = "bus";
-			resets = <&ccu RST_BUS_THS>;
-			nvmem-cells = <&ths_calibration>;
-			nvmem-cell-names = "calibration";
-			#thermal-sensor-cells = <1>;
-		};
-
 		dump_reg: dump_reg@20000 {
 			compatible = "allwinner,sunxi-dump-reg";
 			reg = <0x0 0x03001000 0x0 0x0f20>;
@@ -1516,60 +1511,67 @@ r_lradc: lradc@5070800 {
 		};
 	};
 
+	gpu_opp_table: gpu-opp-table {
+		compatible = "operating-points-v2";
+		opp-125000000 {
+			opp-hz = /bits/ 64 <125000000>;
+			opp-microvolt = <810000>;
+		};
+		opp-250000000 {
+			opp-hz = /bits/ 64 <250000000>;
+			opp-microvolt = <810000>;
+		};
+		opp-432000000 {
+			opp-hz = /bits/ 64 <432000000>;
+			opp-microvolt = <810000>;
+		};
+		opp-600000000 {
+			opp-hz = /bits/ 64 <600000000>;
+			opp-microvolt = <960000>;
+		};
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <1080000>;
+		};
+	};
+
 	thermal-zones {
 		cpu-thermal {
-			polling-delay-passive = <0>;
-			polling-delay = <0>;
+			polling-delay-passive = <500>;
+			polling-delay = <1000>;
 			thermal-sensors = <&ths 2>;
+			sustainable-power = <1000>;
 
 			trips {
-				cpu_alert: cpu-alert {
-					temperature = <85000>;
-					hysteresis = <2000>;
+				cpu_threshold: cpu-trip-0 {
+					temperature = <60000>;
 					type = "passive";
+					hysteresis = <0>;
 				};
-
-				cpu-crit {
-					temperature = <100000>;
+				cpu_target: cpu-trip-1 {
+					temperature = <70000>;
+					type = "passive";
 					hysteresis = <0>;
-					type = "critical";
 				};
-			};
-
-			cooling-maps {
-				map0 {
-					trip = <&cpu_alert>;
-					cooling-device = <&cpu0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
-					                 <&cpu1 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
-					                 <&cpu2 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>,
-					                 <&cpu3 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+				cpu_critical: cpu-trip-2 {
+					temperature = <110000>;
+					type = "critical";
+					hysteresis = <0>;
 				};
 			};
 		};
 
 		gpu-thermal {
-			polling-delay-passive = <0>;
-			polling-delay = <0>;
+			polling-delay-passive = <500>;
+			polling-delay = <1000>;
 			thermal-sensors = <&ths 0>;
+			sustainable-power = <1100>;
 
 			trips {
-				gpu_alert: gpu-alert {
-					temperature = <85000>;
-					hysteresis = <2000>;
-					type = "passive";
-				};
-
-				gpu-crit {
-					temperature = <100000>;
-					hysteresis = <0>;
+				gpu_temp_critical: gpu-trip-0 {
+					temperature = <110000>;
 					type = "critical";
-				};
-			};
-
-			cooling-maps {
-				map0 {
-					trip = <&gpu_alert>;
-					cooling-device = <&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+					hysteresis = <0>;
 				};
 			};
 		};
@@ -1580,10 +1582,10 @@ ve-thermal {
 			thermal-sensors = <&ths 1>;
 
 			trips {
-				ve_alert: ve-alert {
-					temperature = <85000>;
-					hysteresis = <2000>;
-					type = "passive";
+				ve_temp_critical: ve-trip-0 {
+					temperature = <110000>;
+					type = "critical";
+					hysteresis = <0>;
 				};
 			};
 		};
@@ -1594,36 +1596,12 @@ ddr-thermal {
 			thermal-sensors = <&ths 3>;
 
 			trips {
-				ddr_alert: ddr-alert {
-					temperature = <85000>;
-					hysteresis = <2000>;
-					type = "passive";
+				ddr_temp_critical: ddr-trip-0 {
+					temperature = <110000>;
+					type = "critical";
+					hysteresis = <0>;
 				};
 			};
 		};
 	};
-
-	gpu_opp_table: gpu-opp-table {
-		compatible = "operating-points-v2";
-		opp-125000000 {
-			opp-hz = /bits/ 64 <125000000>;
-			opp-microvolt = <810000>;
-		};
-		opp-250000000 {
-			opp-hz = /bits/ 64 <250000000>;
-			opp-microvolt = <810000>;
-		};
-		opp-432000000 {
-			opp-hz = /bits/ 64 <432000000>;
-			opp-microvolt = <810000>;
-		};
-		opp-600000000 {
-			opp-hz = /bits/ 64 <600000000>;
-			opp-microvolt = <960000>;
-		};
-		opp-800000000 {
-			opp-hz = /bits/ 64 <800000000>;
-			opp-microvolt = <1080000>;
-		};
-	};
 };
-- 
2.39.2

